Orange/widgets/visualize/tests/test_owmosaic.py

```diff
@@ -194,3 +194,16 @@ def test_does_not_crash(self):
         data = Table("housing.tab")
         widget.set_data(data)
         vizrank.toggle()
+
+    def test_zerodivision_error(self):
+        """
+        ZeroDivisionError should not be thrown when
+        some kind of invalid data is loaded into widget.
+        GH-2046
+        """
+        a, b = ContinuousVariable("a"), ContinuousVariable("b")
+        table = Table(Domain([a, b]), np.array([[0, np.NaN], [np.NaN, np.NaN]]))
+        try:
+            self.send_signal("Data", table)
+        except ZeroDivisionError:
+            self.assertTrue(False)
```

Comment:
self.fail can be used to make the test fail. And it is also a good idea to specify the reason of the failure ("Widget crashed with ZeroDivisionError" or something like this)